packages/react-import-remote/src/ImportRemote.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import Preconnect from '@shopify/react-preconnect';
+import load from './load';
+
+export interface Props<Imported = any> {
+  source: string;
+  preload?: boolean;
+  onError(error: Error): void;
+  getImport(window: Window): Imported;
+  onImported(imported: Imported): void;
+}
+
+export default class ImportRemote extends React.PureComponent<Props, never> {
+  async componentDidMount() {
+    const {source, getImport, onError, onImported} = this.props;
+
+    try {
+      const imported = await load(source, getImport);
```

Comment:
what about if this is ever rerendered with a different url? usually with these kinds of side-effecty components you need to handle `componentWillUpdate` or `componentDidUpdate`

Reply:
note that render would get called if that happened so we would add a new preconnect but not the script tag. If we used helmet for both we would be able to make them both work the same

Reply:
Yeah that's true. Right now, if this rerenders, then the old Preconnect link is removed (should be?) and the new one is added. At the same point, even if we manually insert script tags, it won't remove the old ones.